### ISSUE TITLE
Challenge-Scheme specific refresh-tokens

### ISSUE THREAD

**Issue by darrenschwarz** [IdentityModel/IdentityModel.AspNetCore#274]
For scenarios where we require management of tokens for multiple resources, we adopt the approach explained in your Duende docs [Requesting access to multiple resources](https://docs.duendesoftware.com/identityserver/v6/fundamentals/resources/isolation/#requesting-access-to-multiple-resources).

There are scenarios where we require the management of access-tokens from different issuers.

I've forked [IdentityModel.AspNetCore](https://github.com/darrenschwarz/IdentityModel.AspNetCore/) made two commits to highlight the requirement and suggested solution.

[Failing test: Succeeds_with_support_for_refresh_token_per_challenge_scheme](https://github.com/darrenschwarz/IdentityModel.AspNetCore/commit/ced535ce6066f5dd7dcda23758b70738c42d7be2)

[Passing test: Succeeds_with_support_for_refresh_token_per_challenge_scheme](https://github.com/darrenschwarz/IdentityModel.AspNetCore/commit/d140de162fae110a02e0c73e120eb980e319370d)

The second commit is really the one of interest where you can see the update to: IdentityModel.AspNetCore.AccessTokenManagement.AuthenticationSessionUserAccessTokenStore.cs

tl;dr; retrieval and persistence of refresh tokens apply a similar treatment as is already applied to the naming of _tokenName_ and _expiresName_; the difference being the ChallengeScheme is appended as opposed to the Resource Name. 

```
var refreshTokenName = $"{TokenPrefix}{OpenIdConnectParameterNames.RefreshToken}";

if (!string.IsNullOrEmpty(parameters.ChallengeScheme))
{
     refreshTokenName += $"::{parameters.ChallengeScheme}";
}
```
Ideally, We'd like to get this change into a PR, pending a review from your side.

I look forward to your feedback. :)

**Comment by leastprivilege** [comment]
merged.
